packages/bigquery/src/dataset.js

```diff
@@ -414,4 +470,15 @@ Dataset.prototype.table = function(id) {
  */
 common.paginator.extend(Dataset, ['getTables']);
 
+/*! Developer Documentation
+ *
+ * All async methods (except for streams) will return a Promise in the event
+ * that a callback is omitted.
+ */
+common.util.promisify(Dataset, {
```

Comment:
Not really sure how that would work? Currently this essentially just wraps all callback style functions to conditionally use promises, pretty much in the same exact manner as `streamRouter`. I think if this were something we extended, we'd probably have to wrap all methods on the fly everytime a class was instantiated..?